### ISSUE TITLE
petrv2     grad_norm nan?

### ISSUE THREAD

**Issue by ningerlove** [megvii-research/PETR#105]

**Comment by yingfei1016** [comment]
Yes, the grad norm always nan or inf, but it doesn't affect performance. 
You can use **optimizer_config = dict(type='Fp16OptimizerHook', loss_scale="dynamic", grad_clip=dict(max_norm=35, norm_type=2))** to make it more stable. Note that the first iter is always inf.

**Comment by ningerlove** [comment]
谢谢您的回答   能解释为啥会出现nan吗 我看到optimizer里梯度和 grad_norm相乘，它为啥不会影响训练

**Comment by exiawsh** [comment]
> 谢谢您的回答 能解释为啥会出现nan吗 我看到optimizer里梯度和 grad_norm相乘，它为啥不会影响训练

mmdet显示的是50个iter里的平均值，这50个iter中只要有一次出现nan或者inf就会log出nan,实际上大部分iter都是正常的，只有个别会出问题，所以对结果影响不大，我是这么理解的。
